server/api.js

```diff
@@ -0,0 +1,551 @@
+/* Conventions for this API:
+ *  - GET method for read actions
+ *  - POST method for create actions (think POSTing a new item to a directory)
+ *  - PUT method for update actions (think PUTting an item over an existing item)
+ *  - DELETE method for delete actions
+ *
+ *  - Must authorize using the Authorization header in all requests
+ *    - Presently, this is done as "Bearer (Google ID Token)" as requests should originate from PA
+ *
+ *  - Must return a JSON object containing the following keys:
+ *    - status:   which will be either "success" or "error"
+ *    - data:     the desired JSON object, if there is one
+ *    - message:  a human-readable explanation of the error, if there was one and this is appropriate. Be careful
+ *                to not include anything that will give an attacker extra information
+ */
+
+// Third-party dependencies
+const Validator = require('express-validator')
+
+// In-house dependencies
+const { helpers } = require('brave-alert-lib')
+const db = require('./db/db')
+
+const validateGetClient = [Validator.header(['Authorization']).notEmpty(), Validator.param(['clientId']).notEmpty()]
+
+async function handleGetClient(req, res) {
+  const validationErrors = Validator.validationResult(req).formatWith(helpers.formatExpressValidationErrors)
+
+  try {
+    if (validationErrors.isEmpty()) {
+      const client = await db.getClientWithId(req.params.clientId)
+
+      // check that the client exists
+      if (client == null) {
```

Comment:
I chose loose equals here just incase the function returns undefined. `helpers.runQuery` will return undefined upon an error, which causes a TypeError when attempting to read `results.rows`, which leads to database functions returning `null` in specific. I think there could be an off-chance that someone messes with a database function, or some extravagant error leads to something returning undefined, that I'd like to catch with this loose equals. Thoughts?

Reply:
Thanks for clarifying! 

I think that seeing `==` in JS code looks funny and will automatically bring up questions. Which means that if you keep it like this, then it would be great if you added a comment with your reasoning.

On the other hand, I think that writing unit tests to guard against any future dev doing funny business with the DB function would be a better safeguard than a loose equals.

I'm OK with either way!